### PR TITLE
CC1352 launchpad support

### PR DESCRIPTION
**Pull request by OYTIS**
Minimal support for CC1352R launchpad and respective MCU. Mostly repeating code for CC26x0, I've tried to factor out the common parts. Errors in unused headers are very likely, although I tried to be careful. No RF yet, sorry.

Wiki page: https://github.com/RIOT-OS/RIOT/wiki/Board:-CC1352-Launchpad

### CODE REVIEW

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_vims.h [hunk 1]
```diff
@@ -155,7 +155,8 @@ typedef struct {
     reg32_t __reserved19[2]; /**< meh */
     reg32_t FSM_BSLP0; /**< FMC FSM bank sector lock program 0 */
     reg32_t FSM_BSLP1; /**< FMC FSM bank sector lock program 1 */
-    reg32_t __reserved20[0x42]; /**< meh */
+    reg32_t FSM_PGM_128; /* Enable 128-bit programming. CC26x2_CC13x2 only*/
+    reg32_t __reserved20[0x41]; /**< meh */
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_vims.h]: glad these `meh`s are still there :-D

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h [hunk 2]
```diff
@@ -34,8 +34,10 @@ typedef struct {
     reg32_t ICR; /**< interrupt clear */
     reg32_t RIS; /**< raw interrupt status */
     reg32_t MIS; /**< masked interrupt status */
+    reg32_t __reserved1[0x100];
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h]: if these two hunks are fixes, i would put them into a separate commit

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 3]
```diff
@@ -43,98 +44,127 @@
...
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
     /* make sure the uart device is valid */
-    if (uart != 0) {
+    if (uart > UART_NUMOF) {
         return UART_NODEV;
     }
 
+#if defined(CPU_VARIANT_X2)
+    uart_regs_t* uart_reg = (uart) ? UART1 : UART0;
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/periph/uart.c]: please use explicit numeric comparisons (`uart != 0`) for numbers.

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 4]
```diff
@@ -43,98 +44,127 @@
...
-    UART->CTL = ENABLE_MASK;
+    uart_reg->CTL = ENABLE_MASK;
 
     return UART_OK;
 }
 
 void uart_write(uart_t uart, const uint8_t *data, size_t len)
 {
-    (void) uart;
+    uart_regs_t* uart_reg = (uart) ? UART1 : UART0;
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/periph/uart.c]: position asterisk next to name (see above)

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 5]
```diff
@@ -43,98 +44,127 @@
...
-        UART->DR = data[i];
+        while (uart_reg->FR & UART_FR_TXFF) {}
+        uart_reg->DR = data[i];
     }
 }
 
 void uart_poweron(uart_t uart)
 {
-    (void) uart;
+    uart_regs_t* uart_reg = (uart) ? UART1 : UART0;
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/periph/uart.c]: position asterisk next to name (see above)
explicit numeric comparison

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 6]
```diff
@@ -43,98 +44,127 @@
...
     while (!(PRCM->CLKLOADCTL & CLKLOADCTL_LOADDONE)) {}
 
-    UART->CTL = ENABLE_MASK;
+    uart_reg->CTL = ENABLE_MASK;
 }
 
 void uart_poweroff(uart_t uart)
 {
-    (void) uart;
+    uart_regs_t* uart_reg = (uart) ? UART1 : UART0;
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/periph/uart.c]: position asterisk next to name (see above)
explicit numeric comparison

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 7]
```diff
@@ -43,98 +44,127 @@
...
     PRCM->UARTCLKGR = 0;
     PRCM->CLKLOADCTL = CLKLOADCTL_LOAD;
     while (!(PRCM->CLKLOADCTL & CLKLOADCTL_LOADDONE)) {}
 
 }
 
-void isr_uart(void)
+static void isr_uart(uart_t uart)
 {
+    uart_regs_t* uart_reg = (uart) ? UART1 : UART0;
```
Comment (yogo1212) [inline cpu/cc26xx_cc13xx/periph/uart.c]: position asterisk next to name (see above)
explicit numeric comparison

**Comment by yogo1212** [comment]
i can't really do much apart from squinting at the code :-/

**Comment by OYTIS** [comment]
Thank you for the review, @yogo1212, I believe everything should be fixed now.
As far as I understand RIOT development process, someone should assign labels to the PR, I don't have enough access rights for that unfortunately.

**Comment by tcschmidt** [comment]
ping @smlng

**Comment by OYTIS** [comment]
@smlng This board is nice as well =) Can split into multiple PRs if convenient.

**Comment by OYTIS** [comment]
@smlng @yogo1212 @tcschmidt  It's been a while since this was first added. Does it make sense for me to rebase, so that we can proceed with the review?

**Comment by tcschmidt** [comment]
This needs testing, I suppose: @smlng ?

**Comment by benpicco** [comment]
@OYTIS yes please rebase.

**Comment by OYTIS** [comment]
@benpicco @smlng @tcschmidt 

Rebased, tested `hello-world` example. Uniflash integration code seems to be out of date, at least it doesn't work with the latest uniflash and the latest xds110 firmware. On the bright side, support for cc13xx/cc26xx has been added to openocd recently, that more or less works.

**Comment by benpicco** [comment]
Murdock found some issues.

**Comment by OYTIS** [comment]
@benpicco OK, CI is happy now.

**Comment by benpicco** [comment]
I'm afraid that as this touches common code, it will have to wait until after the soft feature freeze.
But this looks like it's in a good state so it can be merged once the 2019.10 release branches off.

**Comment by OYTIS** [comment]
@benpicco I can see the 2019.10 in the list of branches. Is it worth rebasing now? Just don't want to do extra work if it doesn't go anywhere.

**Comment by benpicco** [comment]
Yes, please rebase! It's too easy to lose track of PRs on Github…

**Comment by OYTIS** [comment]
@benpicco Thank you! I'm happy that you revived this PR after the long sleep. I'll rebase tomorrow evening.

**Comment by OYTIS** [comment]
@benpicco `codespell` script looks a bit out of control, now it checks itself for typos, and ignores my exception. Not sure what I should do here.

**Comment by OYTIS** [comment]
Hm. I'm not sure why it mattered, but ignoring `codescript/check.sh` worked.

**Comment by benpicco** [comment]
Better make a separate PR for that codespell patch.
Codespell errors don't block merging anyway, that would be super annoying as it sometimes also considers some register names typos.

File: boards/cc1352-launchpad/include/periph_conf.h [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+* to 1 and defining pins for UART_CTS_PIN and UART_RTS_PIN.
+* @{
+*/
+#define UART_NUMOF           (1)
+#define UART0_RX_PIN         (12)
+#define UART0_TX_PIN         (13)
+
+/* Not implemented in launchpad, define if using external transceiver */
+#define UART1_RX_PIN          (0)
+#define UART1_TX_PIN          (0)
```
Comment (benpicco) [inline boards/cc1352-launchpad/include/periph_conf.h]: Can't you just leave those undefined?

File: cpu/cc26x0/include/cpu_conf.h [hunk 9]
```diff
@@ -14,40 +14,24 @@
...
 
-/**
- * @brief   ARM Cortex-M specific CPU configuration
- * @{
- */
-#define CPU_DEFAULT_IRQ_PRIO            (1U)
-#define CPU_IRQ_NUMOF                   IRQN_COUNT
-#define CPU_FLASH_BASE                  FLASH_BASE
-/** @} */
+// Empty, but makes linter happy
```
Comment (benpicco) [inline cpu/cc26x0/include/cpu_conf.h]: The linter will be even happier if you don't use C++ styled comments.

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx.h [hunk 10]
```diff
@@ -28,29 +29,29 @@ extern "C" {
...
+    DebugMonitor_IRQn     = - 4, /**< 12 Cortex-M4 Debug Monitor Interrupt     */
+    PendSV_IRQn           = - 2, /**< 14 Cortex-M4 Pend SV Interrupt           */
+    SysTick_IRQn          = - 1, /**< 15 Cortex-M4 System Tick Interrupt       */
+
+    /******  CC13x2 specific Interrupt Numbers *********************************/
     EDGE_DETECT_IRQN      =   0, /**< 16 AON edge detect                        */
     I2C_IRQN              =   1, /**< 17 I2C                                    */
     RF_CPE1_IRQN          =   2, /**< 18 RF Command and Packet Engine 1         */
-    SPIS_IRQN             =   3, /**< 19 AON SpiSplave Rx, Tx and CS            */
+    PKA_IRQN              =   3, /**< 19 PKA interrupt                          */
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/include/cc26xx_cc13xx.h]: Are those really equal?

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 11]
```diff
@@ -44,50 +45,70 @@
...
-
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
     /* make sure the uart device is valid */
-    if (uart != 0) {
+    if (uart > UART_NUMOF) {
         return UART_NODEV;
     }
 
+#if defined(CPU_VARIANT_X2)
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/periph/uart.c]: I think a `#if defined(UART1_TX_PIN)` would be more generic.
The intent here is that only `CPU_VARIANT_X2` has UART1, right?

Comment (benpicco) [reply cpu/cc26xx_cc13xx/periph/uart.c]:
```C
int tx_pin;
int rx_pin;
switch (uart) {
case 0:
    tx_pin = UART0_TX_PIN;
    …
    break;
#ifdef UART1_TX_PIN
case 1:
    tx_pin = UART1_TX_PIN;
    …
    break;
#endif
}
```

Mind you that other platforms just use a `uart_conf_t` struct and avoid having individual `#ifdefs` for each pin.

File: cpu/cc26xx_cc13xx/periph/uart.c [hunk 12]
```diff
@@ -112,71 +133,84 @@ int uart_mode(uart_t uart, uart_data_bits_t data_bits, uart_parity_t parity,
            stop_bits == UART_STOP_BITS_2);
 
     /* make sure the uart device is valid */
-    if (uart != 0) {
+    if (uart >= UART_NUMOF) {
         return UART_NODEV;
     }
 
+    uart_regs_t *uart_reg = (uart == 1) ? UART1 : UART0;
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/periph/uart.c]: Yea, why not just make
```C
typedef struct {
    uart_regs_t *reg;
    int tx_pin;
    int rx_pin;
    …
} uart_conf_t;
```

and forego the macro madness?

File: cpu/cc26xx_cc13xx/vectors.c [hunk 13]
```diff
@@ -27,21 +28,21 @@ void dummy_handler(void) {
...
+WEAK_DEFAULT void isr_uart0(void);
+WEAK_DEFAULT void isr_aux0_aon(void);
 WEAK_DEFAULT void isr_ssi0(void);
 WEAK_DEFAULT void isr_ssi1(void);
 WEAK_DEFAULT void isr_rfc_cpe0(void);
 WEAK_DEFAULT void isr_rfc_hw(void);
 WEAK_DEFAULT void isr_rfc_cmd_ack(void);
 WEAK_DEFAULT void isr_i2s(void);
-WEAK_DEFAULT void isr_scse1_aon(void);
+WEAK_DEFAULT void isr_aux1_aon(void);
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/vectors.c]: Are those the same?

File: dist/tools/codespell/check.sh [hunk 14]
```diff
@@ -38,7 +38,7 @@ ${CODESPELL_CMD} --version &> /dev/null || {
 CODESPELL_OPTS="-q 2"  # Disable "WARNING: Binary file"
 CODESPELL_OPTS+=" --check-hidden"
 # Disable false positives "nd  => and, 2nd", "WAN => WANT", "od => of"
-CODESPELL_OPTS+=" --ignore-words-list=ND,nd,WAN,od"
+CODESPELL_OPTS+=" --ignore-words-list=ND,nd,WAN,od,DOUT"
```
Comment (benpicco) [inline dist/tools/codespell/check.sh]: Codespell errors won't block a PR, no need to add an exception.

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx.h (hunk 10 above)
Comment (OYTIS) [reply cpu/cc26xx_cc13xx/include/cc26xx_cc13xx.h]: I'm not sure where `SPIS` comes from. [cc13x2/cc26x2 TRM](http://www.ti.com/lit/ug/swcu185d/swcu185d.pdf) states it's PKA (p. 367).

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h [hunk 15]
```diff
@@ -34,8 +34,10 @@ typedef struct {
     reg32_t ICR; /**< interrupt clear */
     reg32_t RIS; /**< raw interrupt status */
     reg32_t MIS; /**< masked interrupt status */
+    reg32_t __reserved1[0x100]; /**< meh */
```
Comment (benpicco) [inline cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h]: Please add a proper commit message.
I guess the memory is really just that fragmented?

File: cpu/cc26xx_cc13xx/vectors.c (hunk 13 above)
Comment (OYTIS) [reply cpu/cc26xx_cc13xx/vectors.c]: Couldn't find SCSE in the datasheet. But it's `AUX software event 1`.

File: dist/tools/codespell/check.sh (hunk 14 above)
Comment (OYTIS) [reply dist/tools/codespell/check.sh]: Sure, that makes sense now.

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h (hunk 15 above)
Comment (OYTIS) [reply cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h]: Ow, sorry, the `meh`s were originally there, I've just copied them for uniformity. Does `reserved` sound better?

**Comment by OYTIS** [comment]
@benpicco `UNIFLASH` support in RIOT doesn't work for me, and so doesn't `compile_and_test_for_board.py`. I used some OpenOCD magic to test, but didn't make it work reliably enough to commit. Do you think it should be in scope for this PR?

**Comment by benpicco** [comment]
A working `make BOARD=cc1352-launchpad flash` should absolutely be in the scope of this PR, otherwise that's just terrible user experience.

Does the tool from #11866 help?

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h (hunk 15 above)
Comment (benpicco) [reply cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h]: I mean this wasn't there in `cc26x0_wdt.h`.
Or do you suggest it should always have been there?

**Comment by OYTIS** [comment]
@benpicco Makes sense to me, I'll see what can be done.

File: cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h (hunk 15 above)
Comment (OYTIS) [reply cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h]: Yes, I've just moved them over (and added my own for new reserved fields).

Comment (benpicco) [reply cpu/cc26xx_cc13xx/include/cc26xx_cc13xx_wdt.h]: I mean they are not your own, they are from the data sheet.

0bd5dbf2-d97e-4ed7-8a69-c2e50a1ee2ae

But yea, they should have been there in the first place and you fixed that.

**Comment by OYTIS** [comment]
@benpicco The timeouts seem to stem from the race conditions really. Like the board is being flashed and spits out the output before `pyterm` can start. Not a lot can be done here in this PR.

File: boards/cc1352-launchpad/include/periph_conf.h [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+     .cfg = GPT_CFG_16T,
+     .chn = 2,
+ },
+ {
+     .cfg = GPT_CFG_32T,
+     .chn = 1,
+ }
+};
+
+#define TIMER_NUMOF         (sizeof(timer_config) / sizeof(timer_config[0]))
```
Comment (benpicco) [inline boards/cc1352-launchpad/include/periph_conf.h]:
```suggestion
#define TIMER_NUMOF         ARRAY_SIZE(timer_config)
```

File: boards/cc1352-launchpad/include/periph_conf.h [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+
+static const uart_conf_t uart_config[] = {
+ {
+     .tx_pin = 13,
+     .rx_pin = 12,
+     .flow_control = 0,
+     .intn = UART0_IRQN
+ }
+};
+#define UART_NUMOF          (sizeof(uart_config)/sizeof(uart_config[0]))
```
Comment (benpicco) [inline boards/cc1352-launchpad/include/periph_conf.h]:
```suggestion
#define UART_NUMOF          ARRAY_SIZE(uart_config)
```

File: boards/cc2650stk/include/periph_conf.h [hunk 18]
```diff
@@ -69,13 +69,20 @@ static const timer_conf_t timer_config[] = {
...
-#define UART_RX_PIN         (28)
-#define UART_TX_PIN         (29)
+
+static const uart_conf_t uart_config[] = {
+ {
+     .tx_pin = 29,
+     .rx_pin = 28,
+     .flow_control = 0,
+     .intn = UART0_IRQN
+ }
```
Comment (benpicco) [inline boards/cc2650stk/include/periph_conf.h]: Yea, C++ does not support leaving gaps or changing the order of the struct members when initializing them like that.
This results in those `sorry, unimplemented: non-trivial designated initializers not supported` errors.

Annoying, I know - I thought C++ was supposed to be more advanced than C?

**Comment by OYTIS** [comment]
@benpicco Apart from race conditions network-related tests fail apparently because the board has no network support. I thought to go through these tests, and add `cc1352-launchpad` to `BOARD_BLACKLIST`, but I can see that blacklists for these tests are empty while there are obviously other boards where network connectivity is not yet supported. Should I leave it as is?

**Comment by benpicco** [comment]
Which tests do you mean?
Usually those network tests should use ethos (Ethernet over Serial) which every board should support.
You might have to run the `tapsetup` script before though - please check the README.md file of the test.

**Comment by OYTIS** [comment]
@benpicco So most of the failures are timeouts. My assumption is that these are because of `pyterm` starting too late. I've run a few select tests manually, and they look good. 

Sometimes you can see that the successful log is there, just some first lines are omitted. Is it the first board that has these issued?
There are a few tests that should be run as root. Do I need to re-run them individually, is that valuable?. There is one test that complains about `Can't find host link-local address on interface tapbr0`. Is it worth digging deeper?

Log is attached.

[test.log](https://github.com/RIOT-OS/RIOT/files/3776604/test.log)

File: cpu/cc13x2/cpu.c [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+    stdio_init();
+
+    /* trigger static peripheral initialization */
+    periph_init();
+}
+
+static void cpu_clock_init(void)
+{
+    /* nothing to be done for cc13x2/cc26x2 */
+}
```
Comment (benpicco) [inline cpu/cc13x2/cpu.c]: If you don't need it just drop that function.

File: cpu/cc13x2/ldscripts/cc1352r1.ld [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+ * @brief           linker script for cc13x2 MCUs
+ *
+ * @}
+ */
+
+/* Memory Space Definitions: */
+MEMORY
+{
+    rom (rx)   : ORIGIN = 0x00000000, LENGTH = 0x58000
+    gpram      : ORIGIN = 0x11000000, LENGTH = 8K /* configurable as cache. 20K here, 8K there, and 2K in the ld-script of cc13x2ware */
```
Comment (benpicco) [inline cpu/cc13x2/ldscripts/cc1352r1.ld]: What does that comment mean?

File: cpu/cc13x2/ldscripts/cc1352r1.ld [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+    ram (w!rx) : ORIGIN = 0x20000000, LENGTH = 20K /* sram */
```
Comment (benpicco) [inline cpu/cc13x2/ldscripts/cc1352r1.ld]: You can define
```Make
ROM_START_ADDR ?= 0x00000000
RAM_START_ADDR ?= 0x20000000

ROM_LEN ?= 352K
RAM_LEN ?= 20K
```

In your cpu's `Makefile.include` and the just use `cortexm_common/ldscripts/cortexm.ld` by defining `LINKER_SCRIPT ?= cortexm.ld`.

File: cpu/cc13x2/ldscripts/cc1352r1.ld (hunk 20 above)
Comment (OYTIS) [reply cpu/cc13x2/ldscripts/cc1352r1.ld]: Honestly, I've just copied it over from cc26x0. I'll take a look into the documentation.

Comment (OYTIS) [reply cpu/cc13x2/ldscripts/cc1352r1.ld]: So I couldn't find anything about ROM/RAM split, but `gpram` can be configured either as cache memory (default) or as additional 8K of RAM. As `gpram` region is not used elsewhere in RIOT, and reconfiguration support code is also missing, I would suggest just removing this line from both cc13x2  and cc26x0 ld scripts.

**Comment by OYTIS** [comment]
@benpicco Linker scripts are fixed now.

**Comment by benpicco** [comment]
Looks good!
You don't happen to have a CC2650 too?

I know I've seen one on the desk today, @miri64 do you know whose that was?
If someone could give this PR a whirl on there too to make sure there are no regressions, this should be good to go.

**Comment by OYTIS** [comment]
@benpicco No, unfortunately I don't. Would be really great to get this tested on cc2650, sure.

**Comment by MrKevinWeiss** [comment]
I can test the cc2650 with this PR.

**Comment by MrKevinWeiss** [comment]
On this PR
```
ERROR:cc2650-launchpad:Tests failed: 11
Failures during compilation:
- [tests/pkg_nanopb](tests/pkg_nanopb/compilation.failed)

Failures during test:
- [tests/bench_msg_pingpong](tests/bench_msg_pingpong/test.failed)
- [tests/driver_grove_ledbar](tests/driver_grove_ledbar/test.failed)
- [tests/driver_my9221](tests/driver_my9221/test.failed)
- [tests/evtimer_underflow](tests/evtimer_underflow/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/gnrc_tcp](tests/gnrc_tcp/test.failed)
- [tests/periph_timer](tests/periph_timer/test.failed)
```

**Comment by OYTIS** [comment]
Hi @MrKevinWeiss,

thank for doing the test! Could you please also run the same test suite for current `master`? I had problems with automated tests for cc1352, just want to make sure it's a degradation, and not a bug in the test runner.

Thanks!

**Comment by MrKevinWeiss** [comment]
Currently undergoing!

**Comment by MrKevinWeiss** [comment]
These things take a while :disappointed_relieved:

**Comment by MrKevinWeiss** [comment]
So on master I got:

```
ERROR:cc2650-launchpad:Tests failed: 9
Failures during compilation:
- [tests/pkg_nanopb](tests/pkg_nanopb/compilation.failed)

Failures during test:
- [tests/driver_grove_ledbar](tests/driver_grove_ledbar/test.failed)
- [tests/driver_my9221](tests/driver_my9221/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/gnrc_tcp](tests/gnrc_tcp/test.failed)
- [tests/periph_timer](tests/periph_timer/test.failed)
```

I will investigate the regressions.

**Comment by MrKevinWeiss** [comment]
`[tests/bench_msg_pingpong](tests/bench_msg_pingpong/test.failed)` I reran manually and it seems to pass and behave the same as master.

**Comment by OYTIS** [comment]
@MrKevinWeiss Thank you!
I _think_ there is a race condition in the test runner, so you will probably get slightly different results on every run.

**Comment by MrKevinWeiss** [comment]
`[tests/evtimer_underflow](tests/evtimer_underflow/test.failed)` I ran manually and it seems to pass.

I think this doesn't introduce any regressions to the cc2560-launchpad.  I didn't run the tests in sudo.  Let me know if you need anything else done on the cc2650-launchpad.

**Comment by MrKevinWeiss** [comment]
> The code looks good and causes no regressions on CC2560.
> 
> Can you please create an issue for the test runner?
> That script should just work, if it reports many false positives, that's pretty bad.

I will just let me verify some things, it may have been the script.

**Comment by MrKevinWeiss** [comment]
Yup, rerunning the full test suite seems pass the master.

**Comment by MrKevinWeiss** [comment]
> Can you please create an issue for the test runner?

I don't know if I should do that.  My USB hub sometimes has some issues, I don't want people to spend time on bugs that are not there.
